src/cmd/initContainer.go

```diff
@@ -0,0 +1,363 @@
+	"os/user"
+	"time"
+
+	"github.com/containers/toolbox/pkg/utils"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	initContFlags struct {
+		home        string
+		homeLink    bool
+		mediaLink   bool
+		mntLink     bool
+		monitorHost bool
+		shell       string
+		uid         int
+		user        string
+	}
+)
+
+var initContainerCmd = &cobra.Command{
+	Use:   "init-container",
+	Short: "Initialize a running container",
+	PreRun: func(cmd *cobra.Command, args []string) {
+		if !utils.PathExists("/run/.containerenv") {
+			logrus.Fatal("The 'init-container' command can only be used inside containers")
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		initContainer(args)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(initContainerCmd)
+	initContainerCmd.Hidden = true
+
+	flags := initContainerCmd.Flags()
+	flags.StringVar(&initContFlags.home, "home", "", "Create a user inside the toolbox container whose login directory is HOME.")
+	initContainerCmd.MarkFlagRequired("home")
+	flags.BoolVar(&initContFlags.homeLink, "home-link", false, "Make /home a symbolic link to /var/home.")
+	flags.BoolVar(&initContFlags.mediaLink, "media-link", false, "Make /media a symbolic link to /run/media")
+	flags.BoolVar(&initContFlags.mntLink, "mnt-link", false, "Make /mnt a symbolic link to /var/mnt")
+	flags.BoolVar(&initContFlags.monitorHost, "monitor-host", false, `Ensure that key configuration files (hosts, timezone,..) inside the toolbox container are in sync with the host.`)
+	flags.StringVar(&initContFlags.shell, "shell", "", "Create a user inside the toolbox container whose login shell is SHELL.")
+	initContainerCmd.MarkFlagRequired("shell")
+	flags.IntVar(&initContFlags.uid, "uid", 0, "Create a user inside the toolbox container whose numerical user ID is UID.")
+	initContainerCmd.MarkFlagRequired("uid")
+	flags.StringVar(&initContFlags.user, "user", "", "Create a user inside the toolbox container whose login name is USER.")
+	initContainerCmd.MarkFlagRequired("user")
+}
+
+func initContainer(args []string) {
+	if viper.GetString("XDG_RUNTIME_DIR") == "" {
+		logrus.Info("XDG_RUNTIME_DIR is unset")
+
+		viper.Set("XDG_RUNTIME_DIR", fmt.Sprintf("/run/user/%d", initContFlags.uid))
+		logrus.Infof("XDG_RUNTIME_DIR set to %s", viper.GetString("XDG_RUNTIME_DIR"))
+	}
+
+	toolboxRuntimeDirectory := fmt.Sprintf("%s/toolbox", viper.GetString("XDG_RUNTIME_DIR"))
+	containerInitializedStamp := fmt.Sprintf("%s/container-initialized-%d", toolboxRuntimeDirectory, os.Getpid())
+
+	logrus.Info("Creating /run/.toolboxenv")
+
+	_, err := os.Create("/run/.toolboxenv")
+	if err != nil {
+		logrus.Fatal("Failed to create /run/.toolboxenv")
+	}
+
+	if initContFlags.monitorHost {
+		logrus.Info("Monitoring host")
+
+		if utils.PathExists("/run/host/etc") {
+			logrus.Info("Path /run/host/etc exists. Mount binding to that location will happen now.")
+
+			err = mountBind("/run/host/etc/machine-id", "/etc/machine-id", "ro")
```

Comment:
With a *real* programming language we ought to be using things like arrays and `for` loops. :)